src/components/organisms/JoinVenueModal/JoinVenueModal.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FC, Fragment, useCallback } from "react";
+import { Modal } from "react-bootstrap";
+import { useHistory } from "react-router-dom";
+
+import { AnyVenue } from "types/Firestore";
+import { venueEntranceUrl, venueInsideUrl } from "utils/url";
+
+import "./JoinVenueModal.scss";
+
+interface JoinVenueModal {
+  venueId?: string;
+  venue: AnyVenue;
```

Comment:
Why both `venueId` and `venue`?

Reply:
Because venue object doesn't have `id` field in it. That's why we are using the `useVenueId` hook everywhere.

Reply:
Can we just make it take a WithId<AnyVenue> and collapse those props together?